core/src/test/java/org/apache/accumulo/core/file/blockfile/cache/TestCachedBlockQueue.java

```diff
@@ -59,16 +110,19 @@ public void testQueue() {
 
     assertEquals(queue.heapSize(), expectedSize);
 
-    LinkedList<org.apache.accumulo.core.file.blockfile.cache.lru.CachedBlock> blocks =
-        queue.getList();
-    assertEquals(blocks.poll().getName(), "cb1");
-    assertEquals(blocks.poll().getName(), "cb2");
-    assertEquals(blocks.poll().getName(), "cb3");
-    assertEquals(blocks.poll().getName(), "cb4");
-    assertEquals(blocks.poll().getName(), "cb5");
-    assertEquals(blocks.poll().getName(), "cb6");
-    assertEquals(blocks.poll().getName(), "cb7");
-    assertEquals(blocks.poll().getName(), "cb8");
+    Iterator<org.apache.accumulo.core.file.blockfile.cache.lru.CachedBlock> blocks =
+       queue.getBlocks().iterator();
+    assertEquals(blocks.next().getName(), "cb1");
+    assertEquals(blocks.next().getName(), "cb2");
+    assertEquals(blocks.next().getName(), "cb3");
+    assertEquals(blocks.next().getName(), "cb4");
+    assertEquals(blocks.next().getName(), "cb5");
+    assertEquals(blocks.next().getName(), "cb6");
+    assertEquals(blocks.next().getName(), "cb7");
+    assertEquals(blocks.next().getName(), "cb8");
+    assertEquals(blocks.next().getName(), "cb9");
+    assertEquals(blocks.next().getName(), "cb10");
+    assertFalse(blocks.hasNext());
```

Comment:
This addition was really needed.  I think this test was not functioning and it passed because it did not have this check.